pkg/proxy/proxy.go

```diff
@@ -47,6 +49,9 @@ type Proxy struct {
 	envoyIntegration *envoyProxyIntegration
 	dnsIntegration   *dnsProxyIntegration
 
+	//PolicyUpdater is the interface to update the policy rules for standalone DNS proxy
+	policyUpdater service.PolicyUpdater
```

Comment:
IMO this shouldn't be a new dependency to the proxy itself - at least it should be part of the `dnsProxyIntegration`

Reply:
Since this is supposed to call the grpc server, should the proxy in that case depend on FqdnDataserver ?

Reply:
No i wonder whether this aspect can be wrapped by the `dnsProxyIntegration` - because "standalone mode" it's part of DNS Proxy integration.

Reply:
I see, so you mean something like this:
```
type dnsProxyIntegration struct {
	dnsProxy       defaultdns.Proxy
	FQDNDataServer *FQDNDataServer
}
```
And then while creating the newDNSproxyIntegration we would have something like this:
```
func newDNSProxyIntegration(dnsProxy defaultdns.Proxy, FQDNDataServer *service.FQDNDataServer) *dnsProxyIntegration {
        ...
	return &dnsProxyIntegration{
		dnsProxy:       dnsProxy,
		FQDNDataServer: FQDNDataServer,
	}
}
```
So in this case, do we need to have `set`/`get` as well ? Not sure becaus, the grpc server is started in the service cell unlike dnsproxy.

Reply:
yes something like that - it's also ok to depend on the interface (using `policyUpdater service.PolicyUpdater`) - just wrap it in the `dnsProxyIntegration`.

AFAIU there's no need to have setters/getters here. why do you think this is necessary? the unexported fields can be accessed from within the proxy package.

Reply:
I wasn't sure if we needed the `setters`/`getters` because was not able to find a reason for it. That's why I was asking.